app/services/task_loader.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class TaskLoader
+  attr_reader :case_record, :workflow_key, :path
+
+  def initialize(case_record, workflow_key)
+    @case_record = case_record
+    @workflow_key = workflow_key
+    @path = Rails.root.join("config", "task_workflows", "#{workflow_key}.yml")
+  end
+
+  def load!
+    unless File.exist?(path)
+      raise ArgumentError, "Tasks workflow file not found: #{workflow_key}"
+    end
+
+    workflow = YAML.load_file(path)
+    raise ArgumentError, "Workflow not found: #{workflow_key}" if workflow.blank?
+
+    build_tasks_for(case_record, workflow)
+  end
```

Comment:
An optimization for later is to preload all the workflows and create loader instances to improve performance